osu.Game.Rulesets.Osu/Mods/OsuModFlashlight.cs

```diff
@@ -1,31 +1,74 @@
 ﻿// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
 // See the LICENCE file in the repository root for full licence text.
 
+using System;
+using System.Collections.Generic;
+using System.Linq;
 using osu.Framework.Bindables;
 using osu.Framework.Graphics;
 using osu.Framework.Input;
 using osu.Framework.Input.Events;
 using osu.Game.Rulesets.Mods;
+using osu.Game.Rulesets.Objects.Drawables;
 using osu.Game.Rulesets.Osu.Objects;
+using osu.Game.Rulesets.Osu.Objects.Drawables;
 using osuTK;
 
 namespace osu.Game.Rulesets.Osu.Mods
 {
-    public class OsuModFlashlight : ModFlashlight<OsuHitObject>
+    public class OsuModFlashlight : ModFlashlight<OsuHitObject>, IApplicableToDrawableHitObjects
     {
         public override double ScoreMultiplier => 1.12;
 
         private const float default_flashlight_size = 180;
 
-        public override Flashlight CreateFlashlight() => new OsuFlashlight();
+        private OsuFlashlight flashlight;
+
+        public override Flashlight CreateFlashlight() => flashlight = new OsuFlashlight();
+
+        public void ApplyToDrawableHitObjects(IEnumerable<DrawableHitObject> drawables)
+        {
+            foreach (var s in drawables.OfType<DrawableSlider>())
+            {
+                s.Tracking.ValueChanged += flashlight.OnSliderTrackingChange;
+            }
+        }
 
         private class OsuFlashlight : Flashlight, IRequireHighFrequencyMousePosition
         {
+            private int trackingSliders;
+
             public OsuFlashlight()
             {
                 FlashlightSize = new Vector2(0, getSizeFor(0));
             }
 
+            public void OnSliderTrackingChange(ValueChangedEvent<bool> e)
+            {
+                // If any sliders are in a tracking state, apply a dim to the entire playfield over a brief duration.
+                if (e.NewValue)
+                {
+                    trackingSliders++;
+                    // This check being here ensures we're only applying a dim if and only if a slider begins tracking.
```

Comment:
you mean "only applying a dim if the *first* slider begins tracking", right? the comment makes little sense right now.

Reply:
The reason why this logic is specifically here where the new value is true is because we probably don't want to be applying a new transition in the case where the number of tracking sliders goes from 2 to 1 (when the new value is false)

Reply:
i was referring to the comment, specifically